### PR TITLE
docs: add APM AWS Lambda release notes

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

EDIT: Reverted most of the changes in this PR. See https://github.com/elastic/apm-aws-lambda/issues/268#issuecomment-1209458338 for the _why_.

- [x] Links the APM AWS Lambda extension release notes to the APM Guide
- [x] Adds the APM AWS Lambda extension to the compatibility table

### Related

* Closes https://github.com/elastic/apm-aws-lambda/issues/242
* Closes https://github.com/elastic/apm-aws-lambda/issues/269
* Related to https://github.com/elastic/apm-aws-lambda/issues/268

### Blockers

- [x] https://github.com/elastic/apm-aws-lambda/pull/270
- [x] https://github.com/elastic/apm-aws-lambda/pull/278

### CODE REVIEW

File: docs/agent-server-compatibility.asciidoc [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
-The chart below outlines the compatibility between different versions of the APM agents and the APM integration/server.
+The chart below outlines the compatibility between different versions of Elastic APM agents and extensions with the APM integration.
 
 [options="header"]
 |====
-|Language |{apm-agent} version |APM integration/server version
+|Language |{apm-agent} version |APM integration version
+// APM AWS Lambda extension
+.1+|**APM AWS Lambda extension**
+|`1.x` |≥ `6.5`
```
Comment (bmorelli25) [inline docs/agent-server-compatibility.asciidoc:11]: @kruskall Do you know what the minimum compatible APM integration/server version is for the APM AWS Lambda extension?

Comment (kruskall) [reply docs/agent-server-compatibility.asciidoc:11]: no, I'll investigate and let you know

Comment (bmorelli25) [reply docs/agent-server-compatibility.asciidoc:11]: 👋 Any update on this? If we're not sure, I'd like to remove the update to the comparison table from this PR so we can merge the release note links.

Comment (axw) [reply docs/agent-server-compatibility.asciidoc:11]: The extension will work with old versions of APM Server, but we only started adding the `faas.*` fields from 8.0.0+. Aside from metrics, they were all there in 8.2.0, so I'd suggest we set that as the minimum version. Full support for Lambda metrics will come in 8.4.0.

Comment (bmorelli25) [reply docs/agent-server-compatibility.asciidoc:11]:
```suggestion
|`1.x` |≥ `8.2`
```

**Pull request merged**